### PR TITLE
General backend improvements

### PR DESCRIPTION
**Pull request by mirkobrombin**
I did a full review of the Bottles backend to avoid the full rewrite. Here some improvements and code clean-up. Will follow a write-up on what needs a rewrite.

### CODE REVIEW

**Comment by francescomasala** [comment]
In my opinion, the if statement in line 94 of the backend/repository.py file can be improved (or it can be removed altogether) since there are no yaml files linked to the bottles version in the mirrors

**Comment by mirkobrombin** [comment]
This is a precaution to serve bottles specific repositories when one or more dependencies/components/programs uses actions not available on older Bottles versions.

**Review by francescomasala — APPROVED** [review]

File: bottles/frontend/utils/connection.py [hunk 1]
```diff
@@ -48,15 +49,23 @@ def check_connection(self, show_notification=False):
             return False
 
         try:
-            urllib.request.urlopen('https://repo.usebottles.com/components/index.yml', timeout=5)
+            res = subprocess.run(
+                ['curl', '-Is', 'https://repo.usebottles.com'],
```
Comment (Kinsteen) [inline bottles/frontend/utils/connection.py:53]: Is curl assumed to be available on every system?

File: bottles/frontend/utils/connection.py [hunk 2]
```diff
@@ -48,15 +49,23 @@ def check_connection(self, show_notification=False):
...
+                stdout=subprocess.DEVNULL,
+                stderr=subprocess.DEVNULL
+            )
+
+            if res.returncode != 0:
+                raise Exception("Connection status: online …")
```
Comment (Kinsteen) [inline bottles/frontend/utils/connection.py:59]: Shouldn't it be "offline", as we're checking if the command failed?

File: bottles/backend/cabextract.py [hunk 3]
```diff
@@ -111,10 +111,7 @@ def __extract(self) -> bool:
                     f"-q {self.path}"
                 ]
                 command = " ".join(command)
-                subprocess.Popen(
-                    command,
-                    shell=True
-                ).communicate()
+                subprocess.Popen(command,  shell=True).communicate()
```
Comment (Kinsteen) [inline bottles/backend/cabextract.py:114]:
```suggestion
                subprocess.Popen(command, shell=True).communicate()
```

File: bottles/backend/health.py [hunk 4]
```diff
@@ -269,6 +269,7 @@ def get_ram_data(self):
 
     def get_results(self, plain: bool = False):
         results = {
+            "Oficial Package": "FLATPAK_ID" in os.environ,
```
Comment (Kinsteen) [inline bottles/backend/health.py:272]:
```suggestion
            "Official Package": "FLATPAK_ID" in os.environ,
```

File: bottles/frontend/utils/connection.py (hunk 1 above)
Comment (mirkobrombin) [reply bottles/frontend/utils/connection.py:53]: It's available in the flatpak which is enough

**Review by pietrodicaprio — APPROVED** [review]

**Pull request merged**